test/Microsoft.TestPlatform.PerformanceTests/PerfInstrumentation/TestPlatformTask.cs

```diff
@@ -27,3 +27,15 @@ public TestPlatformTask(string taskName, double eventStarted)
         PayLoadProperties = new Dictionary<string, string>();
     }
 }
+
+public class TestPlatformEvent
+{
+    public TestPlatformEvent(string eventName, double timeSinceStart)
+    {
+        Name = eventName;
+        TimeSinceStart = timeSinceStart;
+    }
+
+    public string Name { get; }
+    public double TimeSinceStart { get; }
+}
```

Comment:
Nit: Use a record instead

```suggestion
public record TestPlatformEvent(string Name, double TimeSinceStart);
```

Reply:
Yeah that would be nice, probably readonly struct record even.

Reply:
Will skip for now though.